### PR TITLE
fix(accordion): panel-open class not shown with custom class

### LINKED ISSUE #4849: Accordion wont apply panel-open class when initially opened and custom class binding is used
I use the accordion with dynamic class names. I'm using angular bindings for this. As in current version this is not supported via `ng-class` and the `panel-class` attribute is only evaluated once, I have to use the standard `class` attribute. So I do sth like this:

`<accordion-group class={{customClass}} is-open="true"></accordion-group>`

Where `{{customClass}}` will be replaced by `"myCustomClass"`. When the `is-open` attribute is set by default and I navigate to the page with the accordion, the class attribute evaluates to:

`<accordion-group class="myCustomClass panel panel-default panel panel-default" is-open="true"></accordion-group>`

But `"panel-open"` will only be applied if I close and reopen the affected group. You may look this up here and check the class value via debugger (I've just added a constant `class="testClass"` in my example as the result is the same):

http://plnkr.co/edit/cOJDzM2lYdVkA9TWmBDP?p=preview

### PR DESCRIPTION
**Pull request by chenyuzhcy**
Closes #4849 

Working plunker: http://plnkr.co/edit/9ZBDb8jvo0ds5lhTW5MH?p=preview

### CODE REVIEW

**Comment by icfantv** [comment]
@chenyuzhcy, so unfortunately, we can't use `ng-class` here.  In fact, this is what we used to have elsewhere in our code base (and perhaps even here...I don't remember off hand).

The issue is that the `uib-accordion-group` directive uses `replace: true` and Angular has decided to not support reconciling the `ng-class` issues when using this property.

**Comment by wesleycho** [comment]
I don't think we used to use ng-class at all - we did manage to break ng-class support somehow in part due to replace: true being applied.

For reference, here was the prior commit that added the dynamic class addition: https://github.com/angular-ui/bootstrap/commit/5ee23a4c2668e6d1b1a5aef39a1804d7d5d9ec6a#diff-2c7728e12322ff30329ef487569c9435

**Comment by icfantv** [comment]
@chenyuzhcy, [here's the link](https://github.com/angular/angular.js/issues/5695) to the angular issue addressing the `ng-class` issues.

**Comment by icfantv** [comment]
@wesleycho, I couldn't remember exactly, only that we've decided to not support it.  #4847 from a few days ago says as much.

**Comment by wesleycho** [comment]
I thought our decision applied to people using ng-class on the accordion-group element?

**Comment by icfantv** [comment]
...and not the template used internally....yes.  That's totally fair.

File: template/accordion/accordion-group.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="panel {{panelClass || 'panel-default'}}">
```
Comment (wesleycho) [inline template/accordion/accordion-group.html]: I think we do want the ng-class to replace the interpolated class, so
```html
<div class="panel" ng-class="panelClass || 'panel-default'">
```
This is to keep behavior consistent with dynamic class changing.

**Comment by wesleycho** [comment]
I made the change - thanks for spending the time on this quick fix!

**Comment by trickpattyFH20** [comment]
@wesleycho I very passionately disagree with this breaking change to use ng-class.

We had ng-class being used on a accordion-group element which now breaks after upgrading to the latest release.

I belive the ng-class merging should have either been done internally here till Angular releases a fix, or the implementation of ng-class should have just been avoided.

I was going to write a fix for this bug, but since the issue has already been closed I will begrudgingly rewrite my CSS to support ng-class being used on a wrapping parent element, even though this seems ludicrous.

**Comment by wesleycho** [comment]
@trickpattyFH20 you are relying on unsupported behavior in Angular itself with regards to ng-class and directives that use `replace: true`. See #4172 for details.

**Comment by trickpattyFH20** [comment]
@wesleycho wouldn't it be plausible to rework the uibAccordionGroup directive to *not* use replace:true ?

**Comment by wesleycho** [comment]
That would result in breaking how Bootstrap's CSS works, and all theming libraries.

**Comment by trickpattyFH20** [comment]
@wesleycho do you have any sources about what is considered "abuse" of ng-class?
I don't think that an aditional interpolated class should be considered abuse.

I could see it as abuse if you parsed attrs.ngClass and recompiled it in the directive...
but why not just avoid all of that and use the interpolated class like before? This seems like a perfectly reasonable workaround to (#5695)[https://github.com/angular/angular.js/issues/5695]

**Comment by wesleycho** [comment]
A simpler workaround would be to just do `class="{{customClasses()}}"` than bloating library code for a situation explicitly not supported by Angular.

**Pull request closed without merging**